nodebuilder/tests/reconstruct_test.go

```diff
@@ -53,6 +53,9 @@ func TestFullReconstructFromBridge(t *testing.T) {
 	err := bridge.Start(ctx)
 	require.NoError(t, err)
 
+	_, err = bridge.HeaderServ.WaitForHeight(ctx, uint64(blocks))
+	require.NoError(t, err)
+
```

Comment:
```suggestion
  // TODO: This is required to avoid flakes coming from unfinished retry mechanism for the same peer in go-header 
	_, err = bridge.HeaderServ.WaitForHeight(ctx, uint64(blocks))
	require.NoError(t, err)

```

Reply:
Could you also add the same to reconstruct from lights? It has the same issue and is probably very flaky because of that

Reply:
Could you still add TODO to all cases?

Reply:
Removing this Wait will speed up the test, I think

Reply:
345f0e224784308c78d2f4d1c4d28aef220ad458